pkg/internal/checkup/vmi/vmi.go

```diff
@@ -154,20 +154,11 @@ func WithRandomNumberGenerator() Option {
 	}
 }
 
-func WithHugePages() Option {
+func WithHugePages(guestMemory resource.Format) Option {
 	return func(vmi *kvcorev1.VirtualMachineInstance) {
 		vmi.Spec.Domain.Memory = &kvcorev1.Memory{
 			Hugepages: &kvcorev1.Hugepages{PageSize: "1Gi"},
-		}
-	}
-}
-
-func WithMemoryRequest(memory string) Option {
-	return func(vmi *kvcorev1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Resources = kvcorev1.ResourceRequirements{
-			Requests: corev1.ResourceList{
-				corev1.ResourceMemory: resource.MustParse(memory),
-			},
+			Guest:     &resource.Quantity{Format: guestMemory},
```

Comment:
Please consider passing the input value through `resource.MustParse()` so the input value will be validated

https://github.com/kubevirt/kubevirt/blob/053463f177ab7112df50e2bee80f2e688f7621ac/tests/vmi_configuration_test.go#L660